src/app/app.ts

```diff
@@ -66,46 +66,74 @@ const basicSetup: Extension = [
 
 const promqlExtension = new PromQLExtension();
 
-function activateLSPAutocompletion(): void {
-  promqlExtension.setComplete({
-    lsp: {
-      url: 'http://localhost:8080',
-    },
-  });
+function setCompletion() {
+  const completionSelect = document.getElementById('completion') as HTMLSelectElement;
+  const completionValue = completionSelect.options[completionSelect.selectedIndex].value;
+  switch (completionValue) {
+    case 'offline':
+      promqlExtension.setComplete();
+      break;
+    case 'lsp':
+      promqlExtension.setComplete({
+        lsp: {
+          url: 'http://localhost:8080',
+        },
+      });
+      break;
+    case 'prometheus':
+      promqlExtension.setComplete({
+        hybrid: {
+          url: 'http://localhost:9090',
+        },
+      });
+      break;
+    default:
+      promqlExtension.setComplete();
+  }
 }
 
-function activatePrometheusAutocompletion(): void {
-  promqlExtension.setComplete({
-    hybrid: {
-      url: 'http://localhost:9090',
-    },
-  });
+function setLinter() {
+  const completionSelect = document.getElementById('linter') as HTMLSelectElement;
+  const completionValue = completionSelect.options[completionSelect.selectedIndex].value;
+  switch (completionValue) {
+    case 'offline':
+      promqlExtension.setLinter();
+      break;
+    case 'lsp':
+      promqlExtension.setLinter({
+        lsp: {
+          url: 'http://localhost:8080',
+        },
+      });
+      break;
+    default:
+      promqlExtension.setLinter();
+  }
 }
 
-function activateOfflineAutocompletion(): void {
-  promqlExtension.setComplete();
+function createEditor() {
+  new EditorView({
+    state: EditorState.create({ extensions: [basicSetup, promqlExtension.asExtension(), promQLHighlightMaterialTheme] }),
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    parent: document.querySelector('#editor')!,
+  });
 }
 
-new EditorView({
-  state: EditorState.create({ extensions: [basicSetup, promqlExtension.asExtension(), promQLHighlightMaterialTheme] }),
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  parent: document.querySelector('#editor')!,
-});
-
-// eslint-disable-next-line @typescript-eslint/no-non-null-assertion,@typescript-eslint/ban-ts-ignore
-// @ts-ignore
-document.getElementById('lsp').addEventListener('click', function () {
-  activateLSPAutocompletion();
-});
+function applyConfiguration(): void {
+  setCompletion();
+  setLinter();
+  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+  // @ts-ignore
+  // When the linter is changed, it required to reload completely the editor.
+  // So the first thing to do, is to completely delete the previous editor and to recreate it from scratch
+  document.getElementById('editor').innerHTML = '';
```

Comment:
Might be nicer to save the `EditorView` somewhere and call `view.destroy()` on it here vs. just setting the `innerHTML` to empty?

Reply:
yup definitely better. Didn't know it exists ( and also I didn't search it my bad )

Thanks for the info :)